File: packages/shared/src/cognitive/broker-adapter.ts
/**
 * BrokerAdapterReceipt — sello del puerto Paper | Live (ADR-034).
 * PaperBrokerAdapter = venue PAPER. Mock = LIVE not_wired.
 * XtbBrokerAdapter = LIVE vía bridge; submitted ≠ fill; filled→ledger (XL-2).
 * ≠ PaperBroker ≠ PaperOrder ≠ ExecutionRecord.
 */

export type BrokerAdapterVenueV1 = "PAPER" | "LIVE";

export type BrokerAdapterNameV1 = "paper_broker" | "mock" | "xtb";

export type BrokerAdapterFillStatusV1 =
  | "executed"
  | "unknown"
  | "not_wired"
  | "rejected"
  | "submitted";

export type BrokerAdapterReceiptV1 = {
  venue: BrokerAdapterVenueV1;
  adapter: BrokerAdapterNameV1;
  fillStatus: BrokerAdapterFillStatusV1;
};

export const BROKER_ADAPTER_KEY = "brokerAdapter";

export const BROKER_ADAPTER_PAPER = "paper_broker" as const;

export const BROKER_ADAPTER_MOCK = "mock" as const;

export const BROKER_ADAPTER_XTB = "xtb" as const;

export type BuildBrokerAdapterReceiptInputV1 = {
  venue: BrokerAdapterVenueV1;
  adapter: BrokerAdapterNameV1;
  fillStatus: BrokerAdapterFillStatusV1;
};

/** Receipt honesto tras submit por el puerto (paper / mock / xtb). */
export function buildBrokerAdapterReceipt(
  input: BuildBrokerAdapterReceiptInputV1,
): BrokerAdapterReceiptV1 {
  return {
    venue: input.venue,
    adapter: input.adapter,
    fillStatus: input.fillStatus,
  };
}

/** Copy de mesa: LIVE mock ≠ XTB; submitted ≠ fill; filled→ledger (XL-2). */
export function brokerAdapterVenueCopy(
  venue: BrokerAdapterVenueV1,
  adapter?: BrokerAdapterNameV1,
): string {
  if (venue === "LIVE" && adapter === "xtb") {
    return "XTB live — submitted ≠ fill; filled→ledger (≠ paper)";
  }
  if (venue === "LIVE") {
    return "Mock live — no envío (≠ broker live)";
  }
  return "Puerto paper (≠ broker live)";
}

File: packages/py/analytics/src/bolsa_analytics/cognitive/broker_adapter.py
"""BrokerAdapterReceipt — sello del puerto Paper | Live (ADR-034).

PaperBrokerAdapter = venue PAPER. Mock = LIVE not_wired.
XtbBrokerAdapter = LIVE vía bridge; submitted ≠ fill; filled→ledger (XL-2).
≠ PaperBroker ≠ PaperOrder ≠ ExecutionRecord ≠ money path real.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal

BrokerAdapterVenue = Literal["PAPER", "LIVE"]
BrokerAdapterName = Literal["paper_broker", "mock", "xtb"]
BrokerAdapterFillStatus = Literal[
    "executed",
    "unknown",
    "not_wired",
    "rejected",
    "submitted",
]

BROKER_ADAPTER_KEY = "brokerAdapter"
BROKER_ADAPTER_PAPER: BrokerAdapterName = "paper_broker"
BROKER_ADAPTER_MOCK: BrokerAdapterName = "mock"
BROKER_ADAPTER_XTB: BrokerAdapterName = "xtb"


@dataclass(frozen=True, slots=True)
class BrokerAdapterReceipt:
    """Foto honesta del submit. submitted ≠ fill; filled→ledger (XL-2)."""

    venue: BrokerAdapterVenue
    adapter: BrokerAdapterName
    fill_status: BrokerAdapterFillStatus

    def to_dict(self) -> dict[str, object]:
        return {
            "venue": self.venue,
            "adapter": self.adapter,
            "fillStatus": self.fill_status,
        }


def build_broker_adapter_receipt(
    *,
    venue: BrokerAdapterVenue,
    adapter: BrokerAdapterName,
    fill_status: BrokerAdapterFillStatus,
) -> BrokerAdapterReceipt:
    """Receipt tras submit: paper / mock / xtb (submitted ≠ fill; filled→ledger)."""
    return BrokerAdapterReceipt(
        venue=venue,
        adapter=adapter,
        fill_status=fill_status,
    )


def broker_adapter_venue_copy(
    venue: BrokerAdapterVenue,
    adapter: BrokerAdapterName | None = None,
) -> str:
    """Copy de mesa: LIVE mock ≠ XTB; submitted ≠ fill; filled→ledger (XL-2)."""
    if venue == "LIVE" and adapter == "xtb":
        return "XTB live — submitted ≠ fill; filled→ledger (≠ paper)"
    if venue == "LIVE":
        return "Mock live — no envío (≠ broker live)"
    return "Puerto paper (≠ broker live)"
